plugin/src/main/java/net/dzikoysk/funnyguilds/feature/hooks/HookManager.java

```diff
@@ -69,34 +69,34 @@ public void setupHooks() {
             catch (ClassNotFoundException exception) {
                 return null;
             }
-        }).getOrNull();
+        });
 
-        MVDW_PLACEHOLDER_API = setupHook("MVdWPlaceholderAPI", pluginName -> new MVdWPlaceholderAPIHook(pluginName, plugin)).getOrNull();
-        PLACEHOLDER_API = setupHook("PlaceholderAPI", pluginName -> new PlaceholderAPIHook(pluginName, plugin)).getOrNull();
-        LEADER_HEADS = setupHook("LeaderHeads", pluginName -> new LeaderHeadsHook(pluginName, plugin)).getOrNull();
+        MVDW_PLACEHOLDER_API = setupHook("MVdWPlaceholderAPI", pluginName -> new MVdWPlaceholderAPIHook(pluginName, plugin));
+        PLACEHOLDER_API = setupHook("PlaceholderAPI", pluginName -> new PlaceholderAPIHook(pluginName, plugin));
+        LEADER_HEADS = setupHook("LeaderHeads", pluginName -> new LeaderHeadsHook(pluginName, plugin));
 
-        HOLOGRAPHIC_DISPLAYS = this.<FunnyHologramManager>setupHook("HolographicDisplays", pluginName -> new HolographicDisplaysHook(pluginName, plugin))
-                .orElseGet(new EmptyHologramManagerImpl());
+        HOLOGRAPHIC_DISPLAYS = setupHook("HolographicDisplays", pluginName -> new HolographicDisplaysHook(pluginName, plugin),
+                Option.of(new EmptyHologramManagerImpl()), true);
 
-        FUNNY_TAB = setupHook("FunnyTab", pluginName -> new FunnyTabHook(pluginName, plugin), false).getOrNull();
+        FUNNY_TAB = setupHook("FunnyTab", pluginName -> new FunnyTabHook(pluginName, plugin), Option.none(), false);
     }
 
-    public <T> Option<T> setupHook(String pluginName, Function<String, T> hookSupplier, boolean notifyIfMissing) {
+    public <T> Option<T> setupHook(String pluginName, Function<String, T> hookSupplier, Option<T> fallback, boolean notifyIfMissing) {
```

Comment:
Możesz zrobić dodatkową metodę która przyjmuje `T fallback` i może wtedy zwracać nonull

bazowa:
`public <T> Option<T> setupHook(String pluginName, Function<String, T> hookSupplier, boolean notifyIfMissing)`
z fallbackiem
`public <T> T setupHook(String pluginName, Function<String, T> hookSupplier, T fallback, boolean notifyIfMissing)`

i tym fallbackiem były zwracany `EmptyHologramManagerImpl` bez Option
Opakowanie go 2 raz w to jest trochę męczące bo już api ma na optionach

Reply:
Albo pozbyć się `EmptyHologramManagerImpl`, skoro każdy już bazuje na `Option`, to nie ma sensu dostarczać pustych impl dla każdego hooka